lib/src/main/java/com/auth0/msg/AbstractMessage.java

```diff
@@ -0,0 +1,250 @@
+     */
+    public void fromJson(String input) {
+        this.input = input;
+        try {
+            // Convert JSON string to Object
+//            TypeReference<HashMap<String, String>> typeRef
+//                    = new TypeReference<HashMap<String, String>>() {};
+            AbstractMessage msg = mapper.readValue(input, this.getClass());
+            this.claims = msg.getClaims();
+        } catch (JsonGenerationException e) {
+            e.printStackTrace();
+        } catch (JsonMappingException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Takes the claims of this instance of the AbstractMessage class and serializes them
+     * to a json string
+     *
+     * @return a JSON String representation in the form of a hashMap mapping string -> string
+     */
+    public String toJson() throws SerializationException, JsonProcessingException {
+        String jsonMsg = mapper.writeValueAsString(this);
+        if (this.error != null) {
+            throw new InvalidClaimsException("Error present cannot serialize message");
+        }
+        return jsonMsg;
+    }
+
+    /**
+     * @param input the jwt String representation of a message
+     * @param KeyJar that might contain the necessary key
+     */
+    public void fromJwt(String input, KeyJar jar) {
+        this.input = input;
+
+        //This will have logic to parse Jwt to claims
+    }
+
+    /**
+     * Serialize the content of this instance (the claims map) into a jwt string
+     * @param KeyJar the signing keyjar
+     * @param String the algorithm to use in signing the JWT
+     * @return a jwt String
+     * @throws InvalidClaimsException
+     */
+    public String toJwt(KeyJar keyjar, Algorithm algorithm) throws
+            InvalidClaimsException, SerializationException {
+        header.put("alg", algorithm.getName());
+        header.put("typ", "JWT");
+        String signingKeyId = algorithm.getSigningKeyId();
+        if (signingKeyId != null) {
+            header.put("kid", signingKeyId);
+        }
+//        JWTCreator.Builder newBuilder = JWT.create().withHeader(this.header);
+//        for (Claim claimKey: claims.keySet()){
+//            newBuilder.withClaim(claimKey.name, (claimKey.type) claims.get(claimKey));
+//        }
+        return null;
+    }
+
+    /**
+     * Serialize the content of this instance (the claims map) into a jwt string
+     * @param Key the signing key
+     * @param String the algorithm to use in signing the JWT
+     * @return a jwt String
+     * @throws InvalidClaimsException
+     */
+    public String toJwt(Key key, Algorithm algorithm) throws InvalidClaimsException, SerializationException {
+        return null;
+    }
+
+    /**
+     * verify that the required claims are present
+     * @return whether the verification passed
+     */
+    protected boolean verify() {
```

Comment:
The method signature should tell that this method can throw InvalidClaimsException so that caller can appropriately handle failed validation.

Reply:
Done

Reply:
done